dbms/src/Storages/StorageBuffer.cpp

```diff
@@ -241,23 +273,8 @@ class BufferBlockOutputStream : public IBlockOutputStream
         if (!storage.no_destination)
         {
             destination = storage.context.tryGetTable(storage.destination_database, storage.destination_table);
-
-            if (destination)
-            {
-                if (destination.get() == &storage)
-                    throw Exception("Destination table is myself. Write will cause infinite loop.", ErrorCodes::INFINITE_LOOP);
-
-                /// Check table structure.
-                try
-                {
-                    destination->check(block, true);
```

Comment:
I've removed this check to allow writing blocks to `StorageBuffer` which doesn't match the structure of the destination table. All necessary data conversion is done by the function `StorageBuffer::writeBlockToDestination()` just before writing block to the destination table. This change is necessary in case the structure of the destination table is altered.